### PR TITLE
feat: add validation for component/prefab and partial structure

### PR DESCRIPTION
**Pull request by RobinBetty**

### CODE REVIEW

File: src/bb-components-build.ts [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
 import program, { CommanderStatic } from 'commander';
 import { promises, outputJson, pathExists } from 'fs-extra';
-import { ComponentProps, PrefabProps } from './types';
-import { getDuplicateNames, checkRequiredProps } from './utils/validation';
+import { Component, Prefab, Partial } from './types';
+import { validateDuplicateNames } from './utils/validation';
+
+import { validateSchema as validateComponentSchema } from './validations/component';
+import { validateSchema as validatePrefabSchema } from './validations/prefab';
+import { validateSchema as validatePartialSchema } from './validations/partial';
```
Comment (BasBetty) [inline src/bb-components-build.ts]: Is export default possible?

File: tslint.json [hunk 2]
```diff
@@ -29,7 +29,8 @@
     "strict-comparisons": true,
     "static-this": true,
     "no-string-literal": true,
-    "no-floating-promises": true
+    "no-floating-promises": true,
+    "no-floating-promises": false
```
Comment (BasBetty) [inline tslint.json]: Which one is it? :D

File: src/bb-components-build.ts [hunk 3]
```diff
@@ -109,28 +93,28 @@ const buildPartials: (rootDir: string) => Promise<void> = async (
...
 };
 
-Promise.all([
-  buildComponents(rootDir),
-  buildPrefabs(rootDir),
-  buildPartials(rootDir),
-])
-  .then(() => console.info('Built component set.'))
-  .catch(({ message }) => {
+(async () => {
```
Comment (BasBetty) [inline src/bb-components-build.ts]: Can you add a return type, here, and in other place?

File: src/bb-components-build.ts (hunk 1 above)
Comment (larshisken) [reply src/bb-components-build.ts]: We will move some of the other validations (e.g. checking names) to these modules. Also, callers can use `import * as Component from './component'`. So I guess we shouldn't use export default in this case.

**Pull request merged**